core/src/main/scala/org/apache/spark/scheduler/TaskInfo.scala

```diff
@@ -35,12 +35,33 @@ class TaskInfo(
      */
     val index: Int,
     val attemptNumber: Int,
+    /**
+     * The actual RDD partition ID in this task.
+     * The ID of the RDD partition is always same even task retries.
```

Comment:
`The ID of the RDD partition is always same across task or stage retries.`
Also, add that this will be `-1` for historical data, and available for all spark applications since `3.3`.

Reply:
Sure, thanks for your suggestion.